MFEM 4.2 compiles with Hypre up to 2.20, develop is OK.

cessenat opened the pull request:
As reported by @alalazo in https://github.com/spack/spack/pull/24445#pullrequestreview-693701411
MFEM 4.2 does not compile with recent Hypre releases 2.21 or 2.22.
But MFEM git master is OK with these two Hypre releases, introducing a proper hypre test around the 'hypre_ParcsrAdd' call.

cessenat committed:
mfem 4.2 compiles with hypre up to 2.20, develop is OK

cessenat committed:
mfem 4.2 compiles with hypre up to 2.20, develop is OK

cessenat committed:
mfem 4.2 compiles with hypre up to 2.20, develop is OK

cessenat committed:
mfem 4.2 compiles with hypre up to 2.20, develop is OK

adamjstewart committed:
Use flake8-import-order to enforce PEP-8 compliance (#23947)

robertu94 committed:
GDB: resolve warnings about imp being deprecated (#24448)

This patch has already been accepted into gdb's trunk, we just adopting
it earlier here since it is small and gives a better user experience.

michaelkuhn committed:
glib: add v2.68.3 (#24558)

lrknox committed:
Hdf5 cmake (#18937)

* Switch hdf5 package from autotools to cmake.

* Add variant for building with zlib, default to ON.

* Update for format requirements.

* Format change.

* Fix breakage from last merge from develop.
Switch szip to use libaec (unrestricted encryption).
Remove 'static' variant:  static libs will only be installed when
~shared.

* Improve args based on suggestions from pull request.

* Update code URL to github.com
Add/modify 4 depends_on lines to fix running "spack graph --deptype=link hdf5".

* Remove trailing whitespace.

* Remove dependencies added solely to make "spack greph --type=link" work.

* Add new version HDF5 1.8.22.

* Remove unnecessary java_check.

* Fix whitespace for style checks.

* Reverted zlib version dependency to 1.1.2:.
zlib variant removed.
api version default renamed "default".

* Remove blank line.

* Whitespace corrections.

* iRemoved unnecessary 'debug' variant.

* Fix typo in version number in conflict for '+szip'.

* Set default for tools variant to True.
Remove patch functions dependent on 'libtool' file that cmake doesn't
produce.

* Remove line to set ONLY_SHARED_LIBS to true.
Add post_install code to install only one version of tools with shared
linkage and original tool names.

* Remove trailing white space and import of glob package not used.

* Leave BUILD_TESTING set to default which is ON.

* Remove  post_install code to install only one version of tools because
some dependent packages running tests in e4s testing are using
h5diff-shared.  Keep both tools versions for now.

* No longer need to import os.

manuelakuhn committed:
py-scipy: fix missing py-cython dependency (#24548)

Commit:
libzmq: Fix gcc11 build failure (#24563)

cessenat commented:
Stiil the ci failure...

scottwittenburg committed:
pipelines: build warpx on instance with more memory (#24592)

qwertos committed:
New package: py-huggingface-hub (#24588)

* [py-huggingface-hub] created template

* [py-huggingface-hub] added dependencies

* [py-huggingface-hub] added version 0.0.8

* [py-huggingface-hub] Final cleanup

- added description
- added homepage
- removed fixmes

alalazo committed:
vermin: show line numbers of violations (#24580)

This commit runs vermin with the --violations option
that shows details of the violations to target requirements.

viniciusvgp committed:
simgrid: add v3.27, update package (#24513)

holrock committed:
ruby: add v3.0.1 (#24560)

christoph-conrads committed:
CGNS: add CMake dependency (#24564)

qwertos committed:
New package: py-pyautogui (#24572)

* [py-pyautogui] created template

* [py-pyautogui] added some unconditional dependencies

* [py-pyautogui] Final cleanup

- added homepage
- added description
- removed fixmes

* [py-pyautogui] added missing dependencies

v-dobrev commented on var/spack/repos/builtin/packages/mfem/package.py:
```diff
@@ -195,7 +195,8 @@ class Mfem(Package, CudaPackage, ROCmPackage):
     depends_on('mpi', when='+mpi')
     depends_on('hypre@2.10.0:2.13.99', when='@:3.3.99+mpi')
-    depends_on('hypre', when='@3.4:+mpi')
+    depends_on('hypre@:2.20.0', when='@3.4:4.2.0+mpi')
+    depends_on('hypre', when='@4.2.1:+mpi')
```
The next MFEM release will be 4.3, so it makes sense to use:
```suggestion
    depends_on('hypre@:2.20.0', when='@3.4:4.2.99+mpi')
    depends_on('hypre', when='@4.3.0:+mpi')
```

manuelakuhn committed:
py-numpy: conflict with gcc11 and switch master to main (#24573)

Fix syntax of conflict between numpy 1.21.0 and gcc11 to that the clingo
concretizer recognizes it.
In addition the upstream master branch was renamed to main.

haampie committed:
Fix prefix-collision detection for projections (#24049)

If two Specs have the same hash (and prefix) but are not equal, Spack
originally had logic to detect this and raise an error (since both
cannot be installed in the same place). Recently this has eroded and
the check no-longer works; moreover, when defining projections (which
may truncate the hash or other distinguishing properties from the
prefix) Spack was also failing to detect collisions (in both of these
cases, Spack would overwrite the old prefix with the new Spec).

This PR maintains a list of all "taken" prefixes: if a hash is not
registered (i.e. recorded as installed in the database) but the prefix
is occupied, that is a collision. This can detect collisions created
by defining projections (specifically when they omit the hash).

The PR does not detect collisions where specs have the same hash
(and prefix) but are not equal.

manuelakuhn committed:
py-nilearn: add 0.8.0 (#24583)

manuelakuhn committed:
py-nipype: add 1.6.1 (#24584)

adamjstewart committed:
spack style: warn if flake8-import-order is missing (#24590)

skosukhin committed:
hdf5: a follow-up to #18937 (#23820)

qwertos committed:
dependency for py-torchmeta (#24595)

Co-authored-by: Sid Pendelberry <sid@rit.edu>

qwertos committed:
New package: py-datasets (#24597)

* [py-datasets] created template

* [py-datasets] added dependencies

* [py-datasets] requires py-pyarrow with +parquet

* [py-datasets] Final cleanup

- added homepage
- added description
- removed fixmes

ax3l committed:
HiPACE++: FFTW+OpenMP (#24575)

- change the default compute backend: we just start to add OpenMP support
- FFTW: prefer with OpenMP acceleration for OpenMP compute backend

tmadlener committed:
genfit package: add googletest dependency (#24467)

Commit:
darshan-runtime,darshan-util: add v3.3.1 + updated git repository (#24574)

manuelakuhn committed:
r-stringi: add v1.6.2 (#24585)

robert-mijakovic committed:
cmake: add v3.20.4, v3.20.5 (#24582)

Co-authored-by: Robert Mijakovic <robert.mijakovic@lxp.lu>

cessenat committed:
Update var/spack/repos/builtin/packages/mfem/package.py

Co-authored-by: Veselin Dobrev <v-dobrev@users.noreply.github.com>

cessenat committed:
mfem: update import style

cessenat committed:
mfem: update import style

cessenat commented:
Messed up the git story. Closing and shall reopen another PR ; sorry.
